go-controller/pkg/util/util.go

```diff
@@ -329,40 +331,52 @@ func IsClusterIP(svcVIP string) bool {
 	return false
 }
 
-// GetActiveNetworkForNamespace returns the active network for the given namespace
-// based on the NADs present in that namespace.
+func defaultNetConf() *ovncnitypes.NetConf {
+	return &ovncnitypes.NetConf{NetConf: cnitypes.NetConf{Name: types.DefaultNetworkName}}
+}
+
+// FindActiveNetworkForNamespace returns the active network for the given namespace
+// based on the NADs present in that namespace a pointer to NetConf struct.
 // active network here means the network managing this namespace and responsible for
 // plumbing all the entities for this namespace
 // this is:
-// 1) "default" if there are no NADs in the namespace OR all NADs are primaryNetwork:false
-// 2) "<secondary-network-name>" if there is exactly ONE NAD with primaryNetwork:true
-// 3) "unknown" under all other conditions
-func GetActiveNetworkForNamespace(namespace string, nadLister nadlister.NetworkAttachmentDefinitionLister) (string, error) {
-	var activeNetwork string
+// 1) &NetConf{Name: "default"} if there are no NADs in the namespace OR all NADs are primaryNetwork:false
+// 2) &NetConf{Name: "<secondary-network-name>"} if there is exactly ONE NAD with primaryNetwork:true
+// 3) nil under all other conditions
+func FindActiveNetworkForNamespace(namespace string, nadLister nadlister.NetworkAttachmentDefinitionLister) (*ovncnitypes.NetConf, error) {
+	if !IsNetworkSegmentationSupportEnabled() {
+		return defaultNetConf(), nil
+	}
 	namespaceNADs, err := nadLister.NetworkAttachmentDefinitions(namespace).List(labels.Everything())
 	if err != nil {
-		return activeNetwork, err
+		return nil, err
 	}
 	if len(namespaceNADs) == 0 {
-		return types.DefaultNetworkName, nil
+		return defaultNetConf(), nil
 	}
 	numberOfPrimaryNetworks := 0
+	var lastPrimaryNetwork *ovncnitypes.NetConf
```

Comment:
`lastPrimaryNetwork` has been overengineered as variable naming :) My guess you did this because we loop through things so its always saying the last primay network.. :D

let's just call it `primaryNetworkNetConf`